src/lib/statistics/cardinality_estimator.cpp

```diff
@@ -419,12 +419,41 @@ std::shared_ptr<TableStatistics> CardinalityEstimator::estimate_predicate_node(
   // TODO(anybody) Complex predicates are not processed right now and statistics objects are forwarded.
   //               That implies estimating a selectivity of 1 for such predicates.
   if (!operator_scan_predicates) {
-    // We might not have resolved the predicate because it contains subqueries. Unfortunately, we do not know the
-    // values of predicates on uncorrelated subquery results before query execution. However, if the predicate has an
-    // equals or between condition, it acts as a filter comparable to a semi-join with the join key of the subquery
-    // result (see examples below). We obtain such predicates with subquery results from the JoinToPredicateRewriteRule.
-    // This rule also checks that all preconditions are met to ensure correct query results. Thus, we do not check them
-    // here. For more information about this query rewrite, see `join_to_predicate_rewrite_rule.hpp`.
+    // We can obtain predicates with subquery results from the JoinToPredicateRewriteRule, which turns (semi-)joins into
+    // predicates. OperatorScanPredicate::from_expression(...) cannot resolve these predicates. They act as a filter
+    // comparable to a semi-join with the join key of the subquery result (see examples below). The optimizer rule
+    // checks that all preconditions are met to ensure correct query results. Especially, it guarantees that the
+    // subqueries return a single row. Thus, we do not check this here (also, the TableScan operator checks this during
+    // execution). For more information about this query rewrite, see `join_to_predicate_rewrite_rule.hpp`.
+    // In the following, we only check if the predicates look the way they should after the mentioned optimizer rule has
```

Comment:
Add something like "In case other subquery predicates are found which have a different structure as expected, we default to ..." ?
At least I don't see where we state what happens to "normal" subquery predicates.